### DISCUSSION TITLE
Upgrading my opencrx 4.1 project to Opencrx 5.1

### DISCUSSION THREAD

**Discussion by nyemiks** [opencrx/opencrx-documentation#29] [Q&A]
I currently have a custom opencrx 4.1 project that was based on ant.
I want to migrate (upgrade) it to be based on Opencrx 5.1 custom project which is based on gradle.

What are the steps I know to follow to achieve this ?

Note: I have some custom uml models as well.

**Comment by wfro64** [comment]
[here](https://github.com/opencrx/opencrx-custom) you find the custom project `sample` which has been migrated from ant to gradle.

The project dependens on `opencrx-v5.1.0` which is published on maven central. In order to take your openCRX, the following steps are required:
* publish your build on a private / local repository. See task `gradle publish` in `./opencrx/publish/build.gradle.kts`
* modify `./sample/build.gradle.kts` so that it takes the `opencrx` dependencies from this

**Reply by nyemiks** [reply]
I am getting the following error when executing gradle assemble


C:\openmdx_sdk\openmdx_2179\opencrx51\spacio-custom\sample\src\main\java\org\opencrx\sample\backend\Sample.java:74: error: package org.opencrx.sample.account1.jmi1 does not exist
import org.opencrx.sample.account1.jmi1.Gadget;
                                       ^
C:\openmdx_sdk\openmdx_2179\opencrx51\spacio-custom\sample\src\main\java\org\opencrx\sample\backend\Sample.java:75: error: package org.opencrx.sample.client1.jmi1 does not exist
import org.opencrx.sample.client1.jmi1.AccountT;
                                      ^
C:\openmdx_sdk\openmdx_2179\opencrx51\spacio-custom\sample\src\main\java\org\opencrx\sample\backend\Sample.java:76: error: package org.opencrx.sample.client1.jmi1 does not exist
import org.opencrx.sample.client1.jmi1.QueryAccountResult;

task compileJava failed

I modified the following files as follows
1. settings.gradle.kts


 repositories {
	    mavenLocal()
		
        mavenCentral()
        jcenter()
     /**   maven {
            url = uri("https://www.opencrx.org/repos/releases")
        }
		*/
		
    }


I referenced mavenLocal and commented out the opencrx repo

2. build.gradle.kts

repositories {
	mavenLocal()
	
	mavenCentral()
	jcenter()
	
    maven {
        url = uri("https://www.openmdx.org/repos/releases")
    }
/**	
    maven {
        url = uri("https://www.opencrx.org/repos/releases")
    }
*/	
	
}

I commented out the opencrx repo releases.

3. build.properties

target.platform=VERSION_1_8




Before doing the above I navigated to the publish folders in openmdx and opencrx then executed

gradle publish

gradle publishToMavenLocal


Kindly confirm If I modified the files correctly.

**Reply by nyemiks** [reply]
[compileJava_log.txt](https://github.com/opencrx/opencrx-documentation/files/7002594/compileJava_log.txt)
I have attached the complete error log.

It appears to be failing with when trying to compile the sample sources that come with the opencrx-custom project.

**Comment by wfro64** [comment]
I have also had this problem before. Usually another

```
gradle clean
gradle assemble
```

helps.

**Reply by nyemiks** [reply]
Thanks for the response. I will try this and revert.

**Reply by nyemiks** [reply]
It has built successfully. Thanks.
